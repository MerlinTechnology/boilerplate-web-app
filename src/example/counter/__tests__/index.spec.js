import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import { routerMiddleware } from 'react-router-redux'
import { browserHistory } from 'react-router'
import configureMockStore from 'redux-mock-store'

import Counter from '../'
import { initialState } from '../redux'

const middleware = [thunk, routerMiddleware(browserHistory)]
const mockStore = configureMockStore(middleware)

describe('Counter component', () => {
    it('renders correctly', () => {
        const store = mockStore({ counter: initialState })
        const component = renderer.create(
            <Counter store={store} />
        )

        expect(component).toMatchSnapshot()
    })
})
