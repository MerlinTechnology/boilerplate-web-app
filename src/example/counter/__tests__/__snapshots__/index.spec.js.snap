exports[`test renders correctly 1`] = `
<div
  className="main-wrapper">
  <div
    className="main-container row">
    <div
      className="col">
      <h1
        className="h1">
        Counter
      </h1>
      <span
        className="span">
        0
      </span>
      <button
        onClick={[Function]}>
        -
      </button>
      <button
        onClick={[Function]}>
        +
      </button>
    </div>
  </div>
</div>
`;
